src/python/WMCore/Database/CMSCouch.py

```diff
@@ -346,6 +347,39 @@ def updateDocument(self, doc_id, design, update_func, fields={}, useBody=False):
                         (self.name, design, update_func, doc_id)
             return self.put(uri=updateUri, data=fields, decode=False)
 
+    def updateBulkDocuments(self, doc_ids, paramsToUpdate, updateLimits=1000):
+
+        uri = '/%s/_bulk_docs/' % self.name
+        conflictDocIDs = []
+        for ids in grouper(doc_ids, updateLimits):
+            # get original documens
+            docs = self.allDocs(options={"include_docs": True}, keys=ids)['rows']
+            data = {}
+            data['docs'] = []
+            for j in docs:
+                doc = {}
+                doc.update(j['doc'])
+                doc.update(paramsToUpdate)
+                data['docs'].append(doc)
+
+            if len(data['docs']) > 0:
+                retval = self.post(uri, data)
+
+                for idx, result in enumerate(retval):
```

Comment:
If there are several thousands documents and several conflicts, it might be that `idx` will have the same value as previous iterations. How about we store the doc id instead?

Reply:
Idx is doc id. With the post call, it returns doc_id and status.

Reply:
What's the format of `retval`? I insist :), this `idx` will simply be a counter from 0 to len(retval), and, not the doc id.

Reply:
Sorry you're right. I thought I fixed this before. Apparently not. I will update the patch.